dali/operators/reader/loader/video_loader.cc

```diff
@@ -533,6 +533,11 @@ void VideoLoader::read_file() {
     bool is_first_frame = true;
     bool key = false;
     bool seek_must_succeed = false;
+    // how many key frames following the last requested frames we saw so far
+    int key_frames_count = 0;
+    // how many key frames following the last requested frames we need to see before we stop
+    // feeding the decoder
+    const int key_frames_treshold = 2;
```

Comment:
Can we infer this from the file info somehow? I think that there are relatively few formats that support bidirectional prediction - for others, that's just waste of time.
Even in case of HEVC, I _think_ that you don't need the whole interval between keyframes - just from the keyframe to the requested frame (in forward prediction) and from the requested frame to the keyframe (in reverse prediction) - but determining which keyframe is relevant might be non-trivial.

Reply:
I can add check only for HEVC but I'm not aware of any simple way to determine what kind of prediction is used in a given stream - I afraid it would require parsing headers/stream on our own what is much beyond the scope of this task.

> Even in case of HEVC, I think that you don't need the whole interval between keyframes - just from the keyframe to the requested frame (in forward prediction) and from the requested frame to the keyframe (in reverse prediction) - but determining which keyframe is relevant might be non-trivial.

In some cases we don't get any meaningful timestamp from the frame we read from the stream. So it may not work at all.